components/installer/pkg/config/v1/config.go

```diff
@@ -88,6 +88,8 @@ type Config struct {
 
 	SSHGatewayHostKey *ObjectRef `json:"sshGatewayHostKey,omitempty"`
 
+	DisableDefinitelyGP bool `json:"disableDefinitelyGp,omitempty"`
```

Comment:
@corneliusludmann Would it make sense to start this out in `Experimental`? Like having an incubation mode of sorts.

Not particular to this flag, but I fear that we start cluttering the config surface again. And I wonder how many installations make use of this flag...?
Alternatively, we could have sth like `NoOutboundRequests` or `AirGappedMode` or so that could map to the internal `disableDefinitelyGp? :thinking:

Reply:
Glad you ask. Very important points.

**Regarding `experimental`**

I don't see this as an experimental flag because this is needed by customers already. Experimental config values are more values that we use internally but don't expect that users need this. We don't want to tell customers to use experimental flags when they want to install Gitpod in air-gap mode.

**Regarding an `airgap` flag**

I personally would also prefer an `airgap` flag because `disableDefinitelyGp` is pretty technical and for end-users not very clear. As far as I know, @csweichel raised the objection regarding an `airgap` flag that it develops to a magic umbrella flag where no one knows what's behind (maybe he would like to elaborate on this). However ...

**installer vs. Replicated**

... since we would like customers to move to Replicated eventually, they shouldn't have much to do with the installer config. That means we will set the config that is needed for air-gap installations automatically when users run an air-gap installation with Replicated. The high-level user config surface moves to Replicated and it is ok to have more technical flags in the config.

But again, I think it's important to think about it and really appreciate your thoughts on this.  Happy to hear other thoughts as well.